internal/tlsutil/tls.go

```diff
@@ -49,14 +50,26 @@ func NewClientTLS(certFile, keyFile, caFile string) (*tls.Config, error) {
 
 // CertPoolFromFile returns an x509.CertPool containing the certificates
 // in the given PEM-encoded file.
+// For non-windows OS, it loads the system cert pool and adds the certificates
```

Comment:
Wouldn't this be construed as a backwards-incompatible change, and at worst a security issue? There are many valid cases where you would not want Helm to load your system certificate pool.

The use case I'm thinking here... A potential attacker could add a malicious certificate to your root cert chain and redirect you to their chart server. That chart server contains malicious charts, and your Helm client just verified that chart server as a trusted endpoint.

Reply:
I reverted this change back to not loading system certs by default. Maybe we can optionally use system certs. However I will create another proposal/PR for that.